x/auth/ante/unordered_test.go

```diff
@@ -5,140 +5,191 @@ import (
 	"time"
 
 	"github.com/stretchr/testify/require"
+	"go.uber.org/mock/gomock"
 
-	storetypes "cosmossdk.io/store/types"
-
+	"github.com/cosmos/cosmos-sdk/codec"
 	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
-	"github.com/cosmos/cosmos-sdk/runtime"
-	"github.com/cosmos/cosmos-sdk/testutil"
 	"github.com/cosmos/cosmos-sdk/testutil/testdata"
 	sdk "github.com/cosmos/cosmos-sdk/types"
-	moduletestutil "github.com/cosmos/cosmos-sdk/types/module/testutil"
 	"github.com/cosmos/cosmos-sdk/types/tx/signing"
 	"github.com/cosmos/cosmos-sdk/x/auth/ante"
-	authcodec "github.com/cosmos/cosmos-sdk/x/auth/codec"
-	"github.com/cosmos/cosmos-sdk/x/auth/keeper"
-	"github.com/cosmos/cosmos-sdk/x/auth/types"
+	authtx "github.com/cosmos/cosmos-sdk/x/auth/tx"
+	txmodule "github.com/cosmos/cosmos-sdk/x/auth/tx/config"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
 )
 
-func TestUnorderedAnte(t *testing.T) {
-	testPK, _, testAddr := testdata.KeyTestPubAddr()
+func TestSigVerification_UnorderedTxs(t *testing.T) {
+	feeAmount := testdata.NewTestFeeAmount()
+	gasLimit := testdata.NewTestGasLimit()
+	var (
+		suite               *AnteTestSuite
+		priv1, priv2, priv3 cryptotypes.PrivKey
+		antehandler         sdk.AnteHandler
+		defaultSignMode     signing.SignMode
+		accs                []sdk.AccountI
+		msgs                []sdk.Msg
+	)
+	reset := func(isCheckTx, withUnordered bool) {
```

Comment:
Reset feels like a fragile way to do this because it risks maintaining some state from the last test case.

Instead, I would pull everything into a new setup function and suite/fixture struct, ex:
```go
type unorderedTestSuite struct {
                 *AnteTestSuite
                 priv1, priv2, priv3 cryptotypes.PrivKey
		antehandler         sdk.AnteHandler
		defaultSignMode     signing.SignMode
		accs                []sdk.AccountI
		msgs                []sdk.Msg
}

func setupUnorderedTestSuite(isCheck, withUnordered bool) *unorderedTestSuite {
   // ...
}

Reply:
[make the test a suite](https://github.com/cosmos/cosmos-sdk/pull/24573/commits/f36f46f92bbb22f6e191179542b4d0fa38605837)